core/client/hdfs/src/main/java/alluxio/hadoop/LocalCacheFileSystem.java

```diff
@@ -205,4 +206,16 @@ public boolean mkdirs(Path f, FsPermission permission) throws IOException {
   public FileStatus getFileStatus(Path f) throws IOException {
     return mExternalFileSystem.getFileStatus(f);
   }
+
+  @Override
+  public BlockLocation[] getFileBlockLocations(FileStatus file, long start,
+      long len) throws IOException {
+    return mExternalFileSystem.getFileBlockLocations(file, start, len);
```

Comment:
```
// This is used by application to schedule/distribute the tasks
// By returning the real locations in the UFS, we ...
```

Reply:
@dbw9580 @maobaolong what do you think about this?

Reply:
Added it